scripts/o3de/o3de/repo.py

```diff
@@ -295,28 +295,44 @@ def search_repo(manifest_json_data: dict,
                 gem_name: str = None,
                 template_name: str = None,
                 restricted_name: str = None) -> dict or None:
+    o3de_object_uris = []
     if isinstance(engine_name, str) or isinstance(engine_name, pathlib.PurePath):
-        o3de_object_uris = manifest_json_data['engines']
+        try:
+            o3de_object_uris = manifest_json_data['engines']
+        except KeyError:
+            pass
```

Comment:
alternately a bit shorter
```suggestion
        o3de_object_uris = manifest_json_data.get('engines', [])
```